### PR TITLE
Set autoscaling conditions for nodePools

### PR DESCRIPTION
**Pull request by enxebre**

### CODE REVIEW

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 1]
```diff
@@ -215,12 +227,25 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 
 	nodePool.Status.NodeCount = int(wantedMachineSet.Status.AvailableReplicas)
 	if !isAutoscalingEnabled {
+		meta.SetStatusCondition(&nodePool.Status.Conditions, metav1.Condition{
+			Type:   hyperv1.AutoscalingEnabledConditionType,
+			Status: metav1.ConditionFalse,
+			Reason: "asExpected",
```
Comment (ironcladlou) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:233]: pretty sure this should be "AsExpected" for consistency with the rest of the system

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 2]
```diff
@@ -159,6 +165,12 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		if err := validateAutoscalingParameters(nodePool); err != nil {
 			// TODO (alberto): set status conditions:
 			// type: AutoscalingEnabled, status: true/false, reason: asExpected/validationFailed
+			meta.SetStatusCondition(&nodePool.Status.Conditions, metav1.Condition{
+				Type:    hyperv1.AutoscalingEnabledConditionType,
+				Status:  metav1.ConditionFalse,
+				Reason:  "validationFailed",
```
Comment (ironcladlou) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:171]: pretty sure this should be "ValidationFailed" for consistency with the rest of the system

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 3]
```diff
@@ -215,12 +227,25 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
...
 		if nodePool.Status.NodeCount != int(*nodePool.Spec.NodeCount) {
 			log.Info("Requeueing nodePool", "expected available nodes", *nodePool.Spec.NodeCount, "current available nodes", nodePool.Status.NodeCount)
 			return ctrl.Result{Requeue: true}, nil
 		}
 	}
 
+	meta.SetStatusCondition(&nodePool.Status.Conditions, metav1.Condition{
+		Type:    hyperv1.AutoscalingEnabledConditionType,
+		Status:  metav1.ConditionTrue,
+		Reason:  "asExpected",
```
Comment (ironcladlou) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:245]: s/asExpected/AsExpected

File: hypershift-operator/controllers/nodepool/nodepool_controller.go [hunk 4]
```diff
@@ -124,6 +126,10 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	if err != nil {
 		r.Log.Error(err, "Failed to reconcile nodePool")
 		r.recorder.Eventf(nodePool, corev1.EventTypeWarning, "ReconcileError", "%v", err)
+		if err := patchHelper.Patch(ctx, nodePool); err != nil {
```
Comment (ironcladlou) [inline hypershift-operator/controllers/nodepool/nodepool_controller.go:129]: Have you observed this working as expected? Many times now I've seen this utility succeed but not persist anything

Comment (enxebre) [reply hypershift-operator/controllers/nodepool/nodepool_controller.go:129]: yeh, I tested this manually

**Pull request merged**